action-allowedlist/action_allowedlist/actions_parser.py

```diff
@@ -0,0 +1,116 @@
+                    print(f"   Found action used: [{uses}]")
+
+    return actions
+
+def get_all_used_actions(workflow_dir: Path):
+    print("Loading Actions YAML files")
+    if (workflow_dir / ".github/workflows").exists():
+        workflow_files = list((workflow_dir / ".github/workflows").glob("*.yml"))
+    else:
+        workflow_files = list((workflow_dir / "testing-repo/.github/workflows").glob("*.yml"))
+    if not workflow_files:
+        print("Could not find workflow files in the specified directory")
+        return []
+
+    print(f"Found [{len(workflow_files)}] files in the workflows directory")
+    #print(workflow_files)
+    actions_in_repo = []
+    for workflow_file in workflow_files:
+        #print(workflow_file)
+        try:
+            workflow_content = workflow_file.read_text()
+            actions = get_actions_from_file(workflow_content, workflow_file.name)
+            actions_in_repo.extend(actions)
+        except Exception as e:
+            print(f"Error occurred while reading {workflow_file}: {e}")
+
+    return actions_in_repo
+
+def invoke_validate_actions(approved_path, actions_configuration):
+    print("Checking if used actions are approved")
+
+    approved = load_json_file(approved_path)
+    num_approved = 0
+    num_denied = 0
+    num_deprecated = 0
+    unapproved_outputs = []
+    approved_outputs = []
+    found = False
+
+
+    for action in actions_configuration:
+        # print(f"Processing {action['actionLink']} version {action['actionVersion']}") Use for debugging
+
+        approved_action_versions = [a for a in approved if a['actionLink'] == action['actionLink']]
+        if approved_action_versions:
+            # print(f"Approved Versions for {action['actionLink']}: {[a['actionVersion'] for a in approved_action_versions]}") Use for debugging
+            pass
+        else:
+            # print(f"No Approved versions for {action['actionLink']} were found.") Use for debugging
+            pass
+
+        approved_output = [a for a in approved_action_versions if a['actionVersion'] == action['actionVersion']]
+
+        if approved_output:
+            # print(f"Output versions approved: {approved_output}") Use for debugging
+            approved_outputs.append(approved_output[0])
+            num_approved += 1
+
+            # Look for deprecation
+            if approved_output[0].get('deprecated', False):
+                print(f"Using a deprecated version of {action['actionLink']}")
+                num_deprecated += 1
+                found = True
+        else:
+            # print(f"Output versions not approved: {action['actionLink']} version {action['actionVersion']}") Use for debugging
+            unapproved_outputs.append(f"{action['actionLink']} {action['actionVersion']}")
+            num_denied += 1
+            found = True
+
+    if num_denied > 0:
+        e = f"The following {num_denied} actions/versions were denied: {', '.join(unapproved_outputs)}"
+        print(e)
```

Comment:
Suggestion: use a GitHub [error notification](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message)